torch/distributed/fsdp/_shard_utils.py

```diff
@@ -177,21 +177,17 @@ def _create_chunk_dtensor(
     Shard a tensor to chunks along the first dimension. The local rank will gets its
     corresponding chunk as the local tensor to create a DTensor.
     """
-    inner_dim = device_mesh.ndim - 1
-    shard_placement = DShard(0)
-    tensor_list, _ = shard_placement._split_tensor(
-        tensor,
-        device_mesh.size(dim=inner_dim),
-        with_padding=False,
-        contiguous=True,
-    )
-    # We need to explicitly call .clone() here as tensor.chunks() splits a tensor into the specified number of chunks.
-    # Each chunk is a view of the input tensor. If the original tensor change, the view will also be changed.
     # We need to explicitly call .detach() to return a new tensor detached from the current graph.
-    local_tensor = tensor_list[rank].clone().detach()
+    tensor = tensor.clone().detach()
 
     # FSDP placements: [Shard(0)]
     # HSDP placements: [Replicate(), Shard(0)]
-    placements = [Replicate() for _ in range(device_mesh.ndim)]
-    placements[-1] = shard_placement  # type: ignore[call-overload]
-    return DTensor.from_local(local_tensor, device_mesh, placements)
```

Comment:
Do we change the logic of local_tensor here? For the old code, it indicates a local shareded Dtensor. But in the new code, we assume it's always local replicate DTensor?

Reply:
> Do we change the logic of local_tensor here? For the old code, it indicates a local shareded Dtensor. But in the new code, we assume it's always local replicate DTensor?

The input tensor for this function here is actually the global tensor we want to shard. We were sharding it in line 180 - 187 in the previous code path(right above) and we were using `shard_placement._split_tensor()`, which is an internal API inside placement_types.py and we don't want to use it. 

Therefore, instead of doing from_local(local_sharded_tensor, device_mesh, Shard(0)) after splitting the tensor, we are using the public DTensor API to do from_local(global_tensor, device_mesh, Replicate()).redistribute(Shard(0)) to create the DTensor.

Reply:
I see.. If this is the case, why don't we use the api `distribute_tensor`?

https://github.com/pytorch/pytorch/blob/ddb0c265110cfd053009dee34cbc5151bed8cddc/torch/distributed/_tensor/api.py#L424

Reply:
distribute_tensor does not have assumption about layout current tensor passed in, so for replicate it need to broadcast, sharding it need to scatter, we don't want to call collective here. Also since we already know the tensor is replicated, `from_local` marking it as replicate make more sense.

Reply:
@wanchaol Well, I mean this looks like a common DTensor operation, would it be better that we can abstract the common logic somewhere in DTensor or we believe this will only be used here for FSDP?